lecture_notes/extras/curry.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+// Implementation
+
+const curry = (fn) => function curried(...args) {
+  if (fn.length > args.length)
+    return (...args2) => curried(...args, ...args2);
+  else
+    return fn(...args);
+};
```

Comment:
```suggestion
const curry = (fn) => {
  const curried = (...args) => {
    if (fn.length === args.length) return fn(...args);
    return (...args2) => curried(...args, ...args2);
  };
  return curried;
};
```

Reply:
- we do not need else here
- inverting condition we can swap branches
- this solution avoid using `function`, just arrows instead